### PR TITLE
Adding onDragOver eventHandler for grouped list component

### PR DESCRIPTION
**Pull request by pawsing**
#### Description of changes

Added a new event handler onDragOver for GroupedList component. This is needed in order to display proper **visual feedback through cursor** to the user on whether group drop at a particular place is allowed or not. . Currently even for cases where group drop is not allowed the groups can be dragged and the same cursor is shown to the user. We can dynamically update the cursor using DataTransfer.DropEffect in the onDragOver event for which we need onDragOver event in GroupedList component.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12859)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupedListSection.tsx [hunk 1]
```diff
@@ -438,6 +438,7 @@ export class GroupedListSection extends React.Component<IGroupedListSectionProps
       onDragEnter: dragDropEvents!.onDragEnter,
       onDragLeave: dragDropEvents!.onDragLeave,
       onDragEnd: dragDropEvents!.onDragEnd,
+      onDragOver: dragDropEvents!.onDragOver,
```
Comment (sachinma) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupedListSection.tsx:441]: Do you want to make this change to _getRowDragDropOptions in DetailsRowBase class to keep the behavior consistent in groups and rows?

Comment (pawsing) [reply packages/office-ui-fabric-react/src/components/GroupedList/GroupedListSection.tsx:441]: Ok will make this change as well for consistency.

File: packages/office-ui-fabric-react/src/components/Callout/Callout.test.tsx [hunk 2]
```diff
@@ -175,7 +175,7 @@ describe('Callout', () => {
       ReactDOM.render<HTMLDivElement>(
         <div>
           <button id="target" style={{ top: '10px', left: '10px', height: '0', width: '0px' }}>
-target
+            target
```
Comment (sachinma) [inline packages/office-ui-fabric-react/src/components/Callout/Callout.test.tsx:178]: why this change?

File: packages/fluentui/react-northstar/src/components/Table/Table.tsx [hunk 3]
```diff
@@ -58,7 +58,7 @@ export const tableSlotClassNames: TableSlotClassNames = {
   header: `${tableClassName}__header`,
 };
 
-export type TableStylesProps = never
+export type TableStylesProps = never;
```
Comment (sachinma) [inline packages/fluentui/react-northstar/src/components/Table/Table.tsx:61]: This change seems unrelated

File: change/office-ui-fabric-react-2020-04-24-13-39-11-AddingOnDragOverEventForGroupedList.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "minor",
+  "comment": "Adding onDragOverEventHandler for GroupedList component",
```
Comment (sachinma) [inline change/office-ui-fabric-react-2020-04-24-13-39-11-AddingOnDragOverEventForGroupedList.json:3]: You can now change it to say adding onDragOver method to IDragDropEvents..

File: packages/office-ui-fabric-react/src/components/Callout/Callout.test.tsx (hunk 2 above)
Comment (pawsing) [reply packages/office-ui-fabric-react/src/components/Callout/Callout.test.tsx:178]: auto-generated from yarn commands

File: packages/fluentui/react-northstar/src/components/Table/Table.tsx (hunk 3 above)
Comment (pawsing) [reply packages/fluentui/react-northstar/src/components/Table/Table.tsx:61]: auto generated after taking latest pull and running Yarn commands

File: change/office-ui-fabric-react-2020-04-24-13-39-11-AddingOnDragOverEventForGroupedList.json (hunk 4 above)
Comment (pawsing) [reply change/office-ui-fabric-react-2020-04-24-13-39-11-AddingOnDragOverEventForGroupedList.json:3]: Yes that seems nice

**Comment by joschect** [comment]
Any last minute comments? I'm going to force squash and merge unless anyone else speaks up.

**Comment by joschect** [comment]
Retriggering a build since it's been a while, will merge once complete

**Comment by joschect** [comment]
Closing to trigger another build
